### PR TITLE
tidy-html5 4.9.25

### PR DESCRIPTION
**Pull request by rstacruz**
https://github.com/htacg/tidy-html5
http://www.html-tidy.org/

Relevant: #14965

### CODE REVIEW

File: Library/Formula/tidy-html5.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  def install
+    ENV.deparallelize
+    cd "build/cmake"
+    system "cmake", "../..", "-DCMAKE_INSTALL_PREFIX=#{prefix}"
+    system "make"
+    system "make install"
+  end
+
+  test do
+    system bin/"tidy5", "--version"
```
Comment (mistydemeo) [inline Library/Formula/tidy-html5.rb]: Is it possible to write a test that tests the actual functionality?

**Comment by mistydemeo** [comment]
`brew audit` suggests a few revisions:

```
== /usr/local/Library/Formula/tidy-html5.rb ==
C:  1:  9: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
C:  4: 12: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
C:  5:  7: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
C:  8: 14: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

1 file inspected, 4 offenses detected
Error: 5 problems in 1 formulae

==> audit problems
tidy-html5:
 * Stable: version 4.9.25 is redundant with version scanned from URL
 * cmake dependency should be
  depends_on "cmake" => :build
Or if it is indeed a runtime denpendency
  depends_on "cmake" => :run
 * `require 'formula'` is now unnecessary
 * Use `system "make", "install"` instead of `system "make install"` 
 * A top-level "man" directory was found
Homebrew requires that man pages live under share.
This can often be fixed by passing "--mandir=#{man}" to configure.
```

**Comment by jacknagel** [comment]
https://github.com/Homebrew/homebrew-dupes/blob/master/tidy.rb should be addressed if this is merged.

**Comment by mikemcquaid** [comment]
Could you also add a PR to remove https://github.com/Homebrew/homebrew-dupes/blob/master/tidy.rb? Thanks!

**Comment by mikemcquaid** [comment]
A bunch of strict audit things need fixed too: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/24824/version=yosemite/testReport/junit/brew-test-bot/yosemite/audit_tidy_html5___strict/

**Comment by rstacruz** [comment]
Updated it... unfortunately i have no idea how to deal with the man page!

**Comment by mikemcquaid** [comment]
Still failing audit rules: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/24855/version=yosemite/testReport/junit/brew-test-bot/yosemite/audit_tidy_html5___strict/

You can see these locally with `brew audit --strict tidy-html5`

**Comment by rstacruz** [comment]
absolutely—i've no clue how to fix them though, so I'd appreciate some help.

File: Library/Formula/tidy-html5.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class TidyHtml5 < Formula
+  homepage "http://www.html-tidy.org/"
+  url "https://github.com/htacg/tidy-html5/archive/tag-4.9.25.tar.gz"
+
+  depends_on "cmake" => :build
+
+  def install
+    ENV.deparallelize
```
Comment (mikemcquaid) [inline Library/Formula/tidy-html5.rb]: Can you please submit this as an issue to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!

File: Library/Formula/tidy-html5.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    cd "build/cmake"
+    system "cmake", "../..", "-DCMAKE_INSTALL_PREFIX=#{prefix}"
```
Comment (mikemcquaid) [inline Library/Formula/tidy-html5.rb]: Does this need to be done in a subdirectory? Also, use `*std_cmake_args` here.

File: Library/Formula/tidy-html5.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+class TidyHtml5 < Formula
+  homepage "http://www.html-tidy.org/"
+  url "https://github.com/htacg/tidy-html5/archive/tag-4.9.25.tar.gz"
+
```
Comment (mikemcquaid) [inline Library/Formula/tidy-html5.rb]: Needs a `sha256` declared.

File: Library/Formula/tidy-html5.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  depends_on "cmake" => :build
+
+  def install
+    ENV.deparallelize
+    cd "build/cmake"
+    system "cmake", "../..", "-DCMAKE_INSTALL_PREFIX=#{prefix}"
+    system "make"
+    system "make", "install"
+  end
+end
```
Comment (mikemcquaid) [inline Library/Formula/tidy-html5.rb]: Can a test be added to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

File: Library/Formula/tidy-html5.rb [hunk 6]
```diff
@@ -0,0 +1,14 @@
+class TidyHtml5 < Formula
+  homepage "http://www.html-tidy.org/"
+  url "https://github.com/htacg/tidy-html5/archive/tag-4.9.25.tar.gz"
```
Comment (mikemcquaid) [inline Library/Formula/tidy-html5.rb]: Use https://github.com/htacg/tidy-html5/archive/4.9.26.tar.gz which should fix the manpage issue.

File: Library/Formula/tidy-html5.rb (hunk 2 above)
Comment (rstacruz) [inline Library/Formula/tidy-html5.rb]: I just added this defensively not really knowing if it's necessary... it seems it'll do fine with parallel builds.

**Comment by rstacruz** [comment]
Updated. This should fix all outstanding issues.

* now using tidy-html5 4.9.26.
* cmake now uses `*std_make_args`.
* the man issue has been fixed.
* a more thorough test block has been added.

File: Library/Formula/tidy-html5.rb [hunk 7]
```diff
@@ -0,0 +1,21 @@
+class TidyHtml5 < Formula
+  homepage "http://www.html-tidy.org/"
+  url "https://github.com/htacg/tidy-html5/archive/4.9.26.tar.gz"
+  sha256 "28674745db53b6ef1aa4b8466e6e231915dcd596672ec40515d0ab53ee0c33f6"
+
+  depends_on "cmake" => :build
+
+  def install
+    cd "build/cmake"
+    puts std_cmake_args
```
Comment (mistydemeo) [inline Library/Formula/tidy-html5.rb]: Can remove this debug print?

Comment (rstacruz) [inline Library/Formula/tidy-html5.rb]: oops. done

File: Library/Formula/tidy-html5.rb [hunk 8]
```diff
@@ -0,0 +1,20 @@
+class TidyHtml5 < Formula
+  homepage "http://www.html-tidy.org/"
+  url "https://github.com/htacg/tidy-html5/archive/4.9.26.tar.gz"
+  sha256 "28674745db53b6ef1aa4b8466e6e231915dcd596672ec40515d0ab53ee0c33f6"
+
+  depends_on "cmake" => :build
+
+  def install
+    cd "build/cmake"
```
Comment (mikemcquaid) [inline Library/Formula/tidy-html5.rb]: Is it necessary to change directory here? What happens if just run in the root of the repo?

Comment (rstacruz) [inline Library/Formula/tidy-html5.rb]: actually it is, according to tidy's docs. Can confirm: removing that line will cause a build failure.

```diff
  def install
    system "cmake", *std_cmake_args
    system "make"
    system "make", "install"
  end
```

```
==> Downloading https://github.com/htacg/tidy-html5/archive/4.9.26.tar.gz
Already downloaded: /Library/Caches/Homebrew/tidy-html5-4.9.26.tar.gz
==> cmake -DCMAKE_C_FLAGS_RELEASE= -DCMAKE_CXX_FLAGS_RELEASE= -DCMAKE_INSTALL_PREFIX=/usr/local/Cellar/tidy-html5/4.9.26 -DCMAKE_BUILD_T
==> make
warning: failed to load external entity "../documentation/tidy1.xsl"
cannot parse ../documentation/tidy1.xsl
make[2]: *** [man] Error 4
make[1]: *** [CMakeFiles/man.dir/all] Error 2
make: *** [all] Error 2
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1

READ THIS: https://git.io/brew-troubleshooting
```

Comment (mikemcquaid) [inline Library/Formula/tidy-html5.rb]: Weird, thanks.

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**